cli/pkg/publish/publisher.go

```diff
@@ -0,0 +1,115 @@
+package publish
+
+import (
+	"context"
+	"io"
+
+	"github.com/aserto-dev/clui"
+	"github.com/aserto-dev/ds-load/cli/pkg/cc"
+	"github.com/aserto-dev/ds-load/sdk/common/js"
+	"github.com/aserto-dev/ds-load/sdk/common/msg"
+	"github.com/rs/zerolog"
+	"golang.org/x/sync/errgroup"
+
+	dsi "github.com/aserto-dev/go-directory/aserto/directory/importer/v2"
+)
+
+type Publisher struct {
+	UI             *clui.UI
+	Log            *zerolog.Logger
+	importerClient dsi.ImporterClient
+}
+
+func NewPublisher(commonCtx *cc.CommonCtx, importerClient dsi.ImporterClient) *Publisher {
```

Comment:
Should this return a `Publisher` interface?

Reply:
We should always [return a concrete type, not an interface](https://github.com/golang/go/wiki/CodeReviewComments#interfaces). I renamed the struct to DirectoryPublisher so it is different from the interface name.